adobe_vipm/flows/fulfillment/shared.py

```diff
@@ -1034,6 +1043,7 @@ def __call__(self, client, context, next_step):
 
         adobe_client = get_adobe_client()
         one_time_skus = get_one_time_skus(client, context.order)
+        product_id = context.order["agreement"]["product"]["id"]
```

Comment:
I think better to get template here and pass a template to the create payload function. Since we are processing order lines and order belongs to only one product and there is only one template we can save some http queries to the platform